base/math.jl

```diff
@@ -128,15 +128,17 @@ end
 @inline function exp2(x::Base.BitInteger)
     if x > 1023
         Inf64
-    elseif x < -1074
-        Float64(0.0)
     elseif x <= -1023
-        # Result will be a subnormal number
-        reinterpret(Float64, Int64(1) << (x + 1074))
+        if x < -1074
+            # Result will be a subnormal number
+            reinterpret(Float64, (1 % Int64) << (x + 1074))
+        else #-1073 < x <= -1023
+            0.0
+        end
     else
         # If x is a Int128, and is outside the range of Int64, then it is not -1023<x<=1023
         # We will cast everything to Int64 to avoid errors in case of Int128
```

Comment:
I think it is a bit cleaner not to.
Not all paths have the cast, on `x`
only line 141

The cast on the 1 in line 134, is to make sure things work on 32 bit machines where 1 is a 32bit.
Perhaps cleaner, I change that one to  0x0000_0000_0000_0001